python/example/single_cell_allen.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
+
+from collections import defaultdict
+import json
+import arbor
+import seaborn
+import pandas
+import matplotlib.pyplot as plt
+
+# (3) A function that parses the Allen parameter fit file into components for an arbor.decor
+# NB. Needs to be adjusted when using a different model
+def load_allen_fit(fit):
+    with open(fit) as fd:
+        fit = json.load(fd)
+
+    # cable parameters convenience class
+    class parameters:
+        cm = None
+        tempK = None
+        Vm = None
+        rL = None
+
+    param = defaultdict(parameters)
+    mechs = defaultdict(dict)
+    for block in fit['genome']:
+        mech   = block['mechanism'] or 'pas'
+        region = block['section']
+        name   = block['name']
+        value  = float(block['value'])
+        if name.endswith('_' + mech):
+            name = name[:-(len(mech) + 1)]
+        else:
+            if mech == "pas":
+                # transform names and values
+                if name == 'cm':
+                    # scaling factor NEURON -> Arbor
+                    param[region].cm = value/100.0
+                elif name == 'Ra':
+                    param[region].rL = value
+                elif name == 'Vm':
+                    param[region].Vm = value
+                elif name == 'celsius':
+                    param[region].tempK = value + 273.15
+                else:
+                    raise Exception(f"Unknown key: {name}")
+                continue
+            else:
+                raise Exception(f"Illegal combination {mech} {name}")
+        mechs[(region, mech)][name] = value
+
+    regs = [(r, vs) for r, vs in param.items()]
+    mechs = [(r, m, vs) for (r, m), vs in mechs.items()]
+
+    default = parameters()
+    default.tempK=float(fit['conditions'][0]['celsius']) + 273.15
+    default.Vm=float(fit['conditions'][0]['v_init'])
+    default.rL=float(fit['passive'][0]['ra'])
+
+    ions = []
+    for kv in fit['conditions'][0]['erev']:
+        region = kv['section']
+        for k, v in kv.items():
+            if k == 'section':
+                continue
+            ion = k[1:]
+            ions.append((region, ion, float(v)))
+
+    return default, regs, ions, mechs
+
+def make_cell(swc, fit, epas_is_param=True):
```

Comment:
This parameter `epas...` should not be here, it's a bridge between Arbor versions I used to test multiple versions.

Reply:
Sure!